### PR TITLE
Feat/signup tweaks

### LINKED ISSUE #41: sign up tweaks

### PR DESCRIPTION
**Pull request by squigglybob**
resolves #41 

I have moved the signup form to a function in `parts/components.php`  for now as I wanted it to be reusable between different campaign types.

Not sure that's the best naming or place, but that can easily be changed.

The mailchimp subscription is happening in a totally custom way which will save space as the mailchimp validation script is big as it contains jQuery source.

[https://css-tricks.com/form-validation-part-4-validating-mailchimp-subscribe-form/](https://css-tricks.com/form-validation-part-4-validating-mailchimp-subscribe-form/) was reaaallly helpful in terms of how to submit the form correctly

### CODE REVIEW

File: magic-links/24hour/24hour.js [hunk 1]
```diff
@@ -17,6 +17,10 @@ let escapeObject = (obj) => {
   }))
 }
 
+var displayMailChimpStatus = function (data) {
+  console.log(data)
```
Comment (corsacca) [inline magic-links/24hour/24hour.js:21]: ?

**Comment by corsacca** [comment]
@squigglybob can we add the `campaign_subscription` tag to the signups?

File: magic-links/24hour/24hour.js (hunk 1 above)
Comment (squigglybob) [reply magic-links/24hour/24hour.js:21]: it's a debug/getting the response strings out of the mailchimps api response so we could if we wanted to echo out errors/warnings etc. from the mailchimp signup request.

we can get rid of it if we don't want to do that, but hadn't landed one way or the other as wanted to chat it through with you first

Comment (squigglybob) [reply magic-links/24hour/24hour.js:21]: if you watch the console while signing up you'll see them

**Comment by squigglybob** [comment]
you may have to elaborate on what that means 😆

**Comment by squigglybob** [comment]
@corsacca done... i'll merge this one in...

**Pull request merged**